tox.ini

```diff
@@ -1,5 +1,5 @@
 [tox]
-envlist = packaging, py36, py37, py38, py39, check_codestyle, check_isort
+envlist = packaging, py37, py38, py39, check_codestyle, check_isort
```

Comment:
Huh, looks like we somehow missed adding py310 here... I wouldn't balk at sneaking that into this PR

Reply:
Though if I'm honest, I don't understand tox well enough to know what the ramifications of this are. Tests [always seem to use `toxenv: "py"`](https://github.com/matrix-org/synapse/blob/a4521ce0a8d252e77ca8bd261ecf40ba67511a31/.github/workflows/tests.yml#L81-L97), whatever that means

Reply:
Using `py` is an implicit match to the currently running `python` version. Since our CI environment will itself supply the correct Python version for a test run, we don't need to enumerate it separately for Tox.

Where the envlist matters is running tox locally, where you want _Tox_ to handle juggling the requested Python versions.

(E.g., just running `tox` is equivalent to running `tox -e py37,py38,py39,check_codestyle,check_isort`... and which will only succeed if you have Python 3.7, 3.8, and 3.9 available on your local system.)

Reply:
Happy to sneak in py310, will update.